generic/Emscripten-wasm.cmake

```diff
@@ -17,7 +17,15 @@ if(NOT EMSCRIPTEN_PREFIX)
     if(DEFINED ENV{EMSCRIPTEN})
         file(TO_CMAKE_PATH "$ENV{EMSCRIPTEN}" EMSCRIPTEN_PREFIX)
     else()
-        set(EMSCRIPTEN_PREFIX "/usr/lib/emscripten")
+        find_file(_EMCC emcc)
+        if(EXISTS ${_EMCC})
+            get_filename_component(EMSCRIPTEN_PREFIX ${_EMCC} DIRECTORY)
+        else()
+            set(EMSCRIPTEN_PREFIX "/usr/lib/emscripten")
+        endif()
+        unset(_EMCC)
+        #find_file always sets a cache variable. we dont want this
```

Comment:
I *think* having this variable could be a valid use case, in fact -- imagine having two separate Emscripten installations (for whatever reason) and the toolchain picking the wrong one because its `emcc` leaked into `PATH`. Then, to debug this, one could spot the variable being set to an incorrect value (and use `EMSCRIPTEN_PREFIX` next time). A bit bad is that there's now two ways to do the same thing (either setting `EMSCRIPTEN_PREFIX` or this `_EMCC`), but I don't have an immediate idea how to make it better :)

So, alternatively, could the variable be named `_EMSCRIPTEN_EMCC_EXECUTABLE` and this called on it so it's hidden by default but "inspectable" when one needs to?

```cmake
mark_as_advanced(_EMSCRIPTEN_EMCC_EXECUTABLE)
```

Reply:
You're totally right. It slipped my mind that there might be a 'reason' for not being able to disable saving to the cache.